File: src/main/java/com/example/dto/TransformerDto.java
package com.example.dto;

import com.example.entity.mongo.TransformerStatus;

public record TransformerDto(
        Long id,
        String manufacturer,
        String modelType,
        Double ratedPowerKVA,
        Integer primaryVoltageKV,
        Integer secondaryVoltageKV,
        Double frequencyHz,
        Boolean transformerCondition,
        Boolean remoteMonitoring,
        Double currentPower,
        Double currentTemperature,
        Double currentVoltage,
        TransformerStatus status
) {}

File: app/src/main/java/com/example/transformmonitorapp/domain/dto/TransformerDto.kt
package com.example.transformmonitorapp.domain.dto

import com.example.transformmonitorapp.domain.model.TransformerStatus

data class TransformerDto(
    val id: Long?,
    val manufacturer: String?,
    val modelType: String?,
    val ratedPowerKVA: Double?,
    val primaryVoltageKV: Int?,
    val secondaryVoltageKV: Int?,
    val frequencyHz: Double?,
    val transformerCondition: Boolean?,
    val remoteMonitoring: Boolean?,
    val currentPower: Double?,
    val currentTemperature: Double?,
    val currentVoltage: Double?,
    val status: TransformerStatus?
)